Tests/MnemonicSwiftTests.swift

```diff
@@ -1,156 +1,284 @@
+
+        XCTAssertNoThrow(try Mnemonic.validate(mnemonic: twelveWordSeed), "error validating 12 word seed")
+        XCTAssertNoThrow(try Mnemonic.validate(mnemonic: fifteenWordSeed), "error validating 15 word seed")
+        XCTAssertNoThrow(try Mnemonic.validate(mnemonic: eighteenWordSeed), "error validating 18 word seed")
+        XCTAssertNoThrow(try Mnemonic.validate(mnemonic: twentyOneWordSeed), "error validating 21 word seed")
+        XCTAssertNoThrow(try Mnemonic.validate(mnemonic: twentyFourWordSeed), "error validating 24 word seed")
+    }
+
+    func testInvalidWordCount() {
+        let elevenWordSeed = "flash tobacco obey genius army stove desk anchor quarter reflect chalk"
+        let fourteenWordSeed = "figure manual hunt oil unusual outer flee yellow cable bottom uncle okay deputy witness"
+        let seventeenWordSeed = "cost raccoon apple hill success sight bag harvest lawsuit exact snow police camp faith weather squirrel defy"
+        let twentyWordSeed = "beach aim neglect phone boring horror venture door crouch ecology tent bulb oval culture hat half easily crucial horse heart"
+        let twentythreeWordSeed = "vanish dream art  response click orphan patch property owner lawsuit sweet smoke bicycle grunt sentence dish tribe review soap chief soft bone race"
+
+        XCTAssertThrowsError(try Mnemonic.validate(mnemonic: elevenWordSeed))
+        XCTAssertThrowsError(try Mnemonic.validate(mnemonic: fourteenWordSeed))
+        XCTAssertThrowsError(try Mnemonic.validate(mnemonic: seventeenWordSeed))
+        XCTAssertThrowsError(try Mnemonic.validate(mnemonic: twentyWordSeed))
+        XCTAssertThrowsError(try Mnemonic.validate(mnemonic: twentythreeWordSeed))
+    }
+
+    func testApparentlyValidSeedPhraseWithUppercaseCharacter() {
+        let x = "human pulse approve subway climb stairs mind gentle raccoon warfare fog roast sponsor under absorb spirit hurdle animal original honey owner upper empower describe"
+        let y = "Human pulse approve subway climb stairs mind gentle raccoon warfare fog roast sponsor under absorb spirit hurdle animal original honey owner upper empower describe"
+        XCTAssertNoThrow(try Mnemonic.validate(mnemonic: x))// having this nearby makes it easy to look at the code and know exactly whats wrong with y
+        XCTAssertThrowsError(try Mnemonic.validate(mnemonic: y))
+
+        XCTAssertNoThrow(try Mnemonic.deterministicSeedBytes(from: x))
+        XCTAssertThrowsError(try Mnemonic.deterministicSeedBytes(from: y))
+
+    }
+
+    func testSwapTwoWords() {
+        let phrase = "human pulse approve subway climb stairs mind gentle raccoon warfare fog roast sponsor under absorb spirit hurdle animal original honey owner upper empower describe"
+        var x = phrase.components(separatedBy: " ")
+        let i = Int.random(in: 0 ..< x.count)
+        let j = Int.random(in: 0 ..< x.count)
+        x.swapAt(i, j)
```

Comment:
This test will fail even though nothing is wrong if `i` and `j` are the same.

Fix by changing to:

```
let i = Int.random(in: 1 ..< x.count)
let j = Int.random(in: 0 ..< j)
```